tests/testthat/test-set-param.R

```diff
@@ -0,0 +1,443 @@
+      values = 4:6,
+      want = "$theta (1, 4 FIX) 5 6"
+    )
+  )
+  for (case in cases) {
+    ctl <- parse_ctl(c(prob_line, case$lines))
+    set_theta(ctl, case$values)
+    expect_identical(
+      format(ctl),
+      paste0(paste(c(prob_line, case$want), collapse = "\n"), "\n")
+    )
+  }
+})
+
+test_that("set_theta() works: discard bounds", {
+  cases <- list(
+    list(
+      lines = "$theta (1,10,20) 2 3",
+      values = 4:6,
+      want = "$theta 4 5 6"
+    ),
+    list(
+      lines = "$theta (1,10) 2 3",
+      values = 4:6,
+      want = "$theta 4 5 6"
+    ),
+    list(
+      lines = "$theta (1,,10) 2 3",
+      values = 4:6,
+      want = "$theta 4 5 6"
+    ),
+    list(
+      lines = "$theta (1,10 FIX) 2 3",
+      values = 4:6,
+      want = "$theta (4 FIX) 5 6"
+    ),
+    list(
+      lines = "$theta (1, ,20) 2 3",
+      values = 4:6,
+      want = "$theta 4 5 6"
+    ),
+    list(
+      lines = "$theta (1, 10, 20) 2 3",
+      values = 4:6,
+      want = "$theta 4 5 6"
+    ),
+    list(
+      lines = "$theta (1, 10 FIX) 2 3",
+      values = 4:6,
+      want = "$theta (4 FIX) 5 6"
+    ),
+    list(
+      lines = c(
+        "$theta (1, 2, 3)",
+        "$table time",
+        "$theta (4, 5, 6) (7, 8, 9)"
+      ),
+      values = c(20, NA, 80),
+      want = c(
+        "$theta 20",
+        "$table time",
+        "$theta (4, 5, 6) 80"
+      )
+    )
+  )
+  for (case in cases) {
+    ctl <- parse_ctl(c(prob_line, case$lines))
+    set_theta(ctl, case$values, bounds = "discard")
+    expect_identical(
+      format(ctl),
+      paste0(paste(c(prob_line, case$want), collapse = "\n"), "\n")
+    )
+  }
+})
+
+test_that("set_omega() works", {
+  cases <- list(
+    list(
+      lines = "$omega 1 2 3",
+      values = c(
+        4,
+        0, 5,
+        0, 0, 6
+      ),
+      want = "$omega 4 5 6"
+    ),
+    list(
+      lines = "$omega 1 2 3 SD",
+      values = matrix(
+        c(
+          4, 0, 0,
+          0, 5, 0,
+          0, 0, 6
+        ),
+        nrow = 3, byrow = TRUE,
+      ),
+      want = "$omega 4 5 6 SD"
+    ),
+    list(
+      lines = c(
+        "$omega BLOCK(2)",
+        "1.1 ; c1",
+        "1.2  1.3",
+        "$table time",
+        "$omega BLOCK(3) VALUES(1.4, 1.5) ; diag, odiag"
+      ),
+      values = matrix(
+        c(
+          2, 0, 0, 0, 0,
+          3, 4, 0, 0, 0,
+          0, 0, 5, 0, 0,
+          0, 0, 6, 0, 0,
+          0, 0, 0, 0, 0
+        ),
+        nrow = 5, byrow = TRUE,
+      ),
+      want = c(
+        "$omega BLOCK(2)",
+        "2 ; c1",
+        "3  4",
+        "$table time",
+        "$omega BLOCK(3) VALUES(5, 6) ; diag, odiag"
+      )
+    ),
+    list(
+      lines = c(
+        "$omega 1.1",
+        "$omega BLOCK(3) VALUES(1.2, 1.3)"
+      ),
+      values = matrix(
+        c(
+          2, 0, 0, 0,
+          0, 3, 0, 0,
+          0, NA, 0, 0,
+          0, 0, 0, 0
+        ),
+        nrow = 4, byrow = TRUE,
```

Comment:
Perhaps I missed a commit message or docs, but I was just wondering what the motivation was for adding support for skipping values via `NA` was. It's nice to have that functionality in `nmrec`, but did you envision a way to implement this in `bbr`?

Reply:
It seemed like a natural and straightforwad way to support updating only a subset of values for callers that are using these functions directly.

I didn't envision bbr exposing this functionality.